backend/danswer/db/engine.py

```diff
@@ -295,30 +295,32 @@ async def get_async_session_with_tenant(
 def get_session_with_tenant(
     tenant_id: str | None = None,
 ) -> Generator[Session, None, None]:
-    """Generate a database session with the appropriate tenant schema set."""
+    """Generate a database session bound to a connection with the appropriate tenant schema set."""
     engine = get_sqlalchemy_engine()
+
     if tenant_id is None:
         tenant_id = current_tenant_id.get()
+    else:
+        current_tenant_id.set(tenant_id)
```

Comment:
Can we name or organize "current_tenant_id" in such a way that the fact it's a context var is more obvious?

Reply:
Great point - I think renaming is a good idea– seems worth putting in this PR